lib/iris/tests/experimental/regrid/test_regrid_area_weighted_rectilinear_src_and_grid.py

```diff
@@ -322,19 +322,36 @@ def test_regrid_latlon_reduced_res(self):
         res = regrid_area_weighted(src, dest)
         self.assertCMLApproxData(res, RESULT_DIR + ("latlonreduced.cml",))
 
-    def test_regrid_transposed(self):
-        src = self.simple_cube.copy()
-        dest = _subsampled_grid(src, 2, 3)
-        # Transpose src so that the coords are not y, x ordered.
-        src.transpose()
+    def test_regrid_reorder_axis(self):
+        src = self.realistic_cube[0, :4, :3, :2]
+        z = src.coord("model_level_number")
+        lat = src.coord("grid_latitude")
+        lon = src.coord("grid_longitude")
+        dest = _resampled_grid(self.realistic_cube[0, 0, :3, :2], 3, 3)
         res = regrid_area_weighted(src, dest)
-        self.assertCMLApproxData(res, RESULT_DIR + ("trasposed.cml",))
-        # Using original and transposing the result should give the
-        # same answer.
-        src = self.simple_cube.copy()
+        self.assertArrayShapeStats(src, (4, 3, 2), 288.08868, 0.008262919)
+        self.assertArrayShapeStats(res, (4, 9, 6), 288.08865, 0.00826281)
+        # Reshape src so that the coords are ordered [x, z, y],
+        # the mean and std statistics should be the same
+        data = np.moveaxis(src.data.copy(), 2, 0)
+        src = iris.cube.Cube(data)
+        src.add_dim_coord(lat, 2)
+        src.add_dim_coord(z, 1)
+        src.add_dim_coord(lon, 0)
         res = regrid_area_weighted(src, dest)
-        res.transpose()
-        self.assertCMLApproxData(res, RESULT_DIR + ("trasposed.cml",))
+        self.assertArrayShapeStats(src, (2, 4, 3), 288.08868, 0.008262919)
+        self.assertArrayShapeStats(res, (6, 4, 9), 288.08865, 0.00826281)
+        # Reshape src so that the coords are ordered [y, x, z],
+        # the mean and std statistics should be the same
+        data = np.moveaxis(src.data.copy(), 2, 0)
+        src = iris.cube.Cube(data)
+        src.add_dim_coord(z, 2)
+        src.add_dim_coord(lon, 1)
+        src.add_dim_coord(lat, 0)
+        dest = _resampled_grid(self.realistic_cube[0, 0, :3, :2], 3, 3)
+        res = regrid_area_weighted(src, dest)
+        self.assertArrayShapeStats(src, (3, 2, 4), 288.08868, 0.008262919)
+        self.assertArrayShapeStats(res, (9, 6, 4), 288.08865, 0.00826281)
```

Comment:
Should these different tests be in separate test functions?

Reply:
Usually yes, but in this case they align with the transpose function that this supersedes and also share the same results info.